app/controllers/posts_controller.rb

```diff
@@ -1,5 +1,10 @@
 class PostsController < ApplicationController
-  def index; end
+  def index
+    @user = User.find(params[:user_id])
+  end
 
-  def show; end
+  def show
+    @user = User.find(params[:user_id])
+    @post = Post.find(params[:id])
+  end
```

Comment:
- [ ] As you may notice, once creating a user, It prompts me to insert the `post_counter` parameter as well which is supposed to have a default of `0` in your database and be incremented as the user is posting on the platform, Kindly make sure you correct that in your `User Models`
![image](https://github.com/kwesiObuobi/blog_app/assets/97100091/4aab1342-8abd-4bc1-8494-92c035d286ca)

hint: You may consider saying for example:
```rb
class User < ApplicationRecord
  #has_many :posts, foreign_key: :author_id
  #has_many :comments, foreign_key: :author_id
  #has_many :likes, foreign_key: :author_id

  #validates :name, presence: true
  #validates :posts_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
  after_initialize :initialize_post-counter
  before_create :set_default_avatar #here you can consider adding a fx that will create a default image avatar when the user is created or you can leave it for now as you have the URL provided for now

  #scope :three_recent_posts, ->(user) { user.posts.order(created_at: :desc).limit(3) }
#end
def initialize_post-counter #This method aims to initialize the number of post to zero once a post is created if the user haven't yet post anything.
    self.posts_count ||= 0
    true
  end
```

Reply:
- [ ] For the comment you will need to do the same thing

Reply:
you're welcome @kwesiObuobi